### PR TITLE
Extract OSGI configuration to new package ui.config

### PR DESCRIPTION
**Pull request by SzymonSzafraniec**
Corresponding issue: https://github.com/Adobe-Consulting-Services/acs-aem-commons/issues/3433

Currently acs-aem-commons-all package mixes subpackages with configuration files. This change bundles the configurations in a separate package which is also included in the all package.

Another benefit is that it allows other projects to include individual ACS Commons packages while not missing the OSGI configurations (e.g. include the acs-aem-commons-bundle and acs-aem-commons-ui.config only). 

Also, by including individual packages separately it's possible to control the packages installation order. This is currently impossible since including the acs-aem-commons-all is the only way to install OSGI Configurations.

### CODE REVIEW

**Comment by davidjgonzalez** [comment]
@kwin  any problems with this?

File: ui.config/pom.xml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <!-- ====================================================================== -->
+    <!-- P A R E N T P R O J E C T D E S C R I P T I O N -->
+    <!-- ====================================================================== -->
+    <parent>
+        <groupId>com.adobe.acs</groupId>
+        <artifactId>acs-aem-commons</artifactId>
+        <version>6.6.5-SNAPSHOT</version>
```
Comment (kwin) [inline ui.config/pom.xml:28]: This is the wrong version and therefore does build.

File: ui.config/pom.xml [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        </plugins>
+    </build>
+    <dependencies>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>acs-aem-commons-oakpal-checks</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Comment (kwin) [inline ui.config/pom.xml:133]: I don't think this requires logically any dependencies

**Review by kwin — CHANGES_REQUESTED** [review]
Looks good in general, just requires some update due to the latest merge.

File: ui.config/pom.xml (hunk 1 above)
Comment (davidjgonzalez) [inline ui.config/pom.xml:28]:
```suggestion
        <version>6.7.0-SNAPSHOT</version>
```

File: ui.config/pom.xml [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            <artifactId>acs-aem-commons-bundle</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <!-- put UberJar last so that more specific artifacts take precedence -->
+        <dependency>
+            <groupId>com.adobe.aem</groupId>
+            <artifactId>uber-jar</artifactId>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
```
Comment (davidjgonzalez) [inline ui.config/pom.xml:144]:
```suggestion
```

File: ui.config/pom.xml [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        <plugins>
+            <plugin>
+                <groupId>org.apache.jackrabbit</groupId>
+                <artifactId>filevault-package-maven-plugin</artifactId>
+                <configuration>
+                    <packageType>container</packageType>
+                </configuration>
+            </plugin>
+
+            <plugin>
```
Comment (kwin) [inline ui.config/pom.xml:51]: I think filevault validation is enough for this package.

File: ui.config/pom.xml [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>license-maven-plugin</artifactId>
+                <configuration>
+                    <excludes>
+                        <exclude>**/*.xml</exclude>
+                    </excludes>
+                </configuration>
+            </plugin>
```
Comment (davidjgonzalez) [inline ui.config/pom.xml:123]:
```suggestion           
```

**Pull request merged**